### PR TITLE
feat: add helm chart releasing

### PR DESCRIPTION
**Pull request by morremeyer**
**How to categorize this PR?**
/area usability
/kind enhancement

**What this PR does / why we need it**:
This implements the chart releasing and documentation for it as discussed in #4123.

**Which issue(s) this PR fixes**:
Parts of #4123 

**Special notes for your reviewer**:

**Release note**:

```feature operator
helm charts are now released in a repository. 
```

### CODE REVIEW

**Comment by gardener-robot-ci-2** [comment]
Thank you @morremeyer for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

File: docs/development/process.md
Comment (rfranzke) [inline docs/development/process.md:49]: This should be automated via a `prepare_release` step in the CI/CD system. No human will do it manually. cc @ccwienk

Comment (morremeyer) [reply docs/development/process.md:49]: I agree! I think I can pull the version from the VERSION file, is that correct?

Comment (rfranzke) [reply docs/development/process.md:49]: I think you can get some inspiration from https://github.com/gardener/gardener/blob/e84d5ab32a96b722284e89252ce992615ec4f600/hack/.ci/prepare_release, namely https://github.com/gardener/gardener/blob/e84d5ab32a96b722284e89252ce992615ec4f600/hack/.ci/prepare_release#L71. This script is used in `gardener-extension-*` repositories to auto-generate the `example/controller-registration.yaml` files during a new release (e.g. https://github.com/gardener/gardener-extension-provider-aws/commit/6ae742f41b8c9d1bf03d7db2b234e002044ae702). Without such script, the only change during a "release commit" is bumping the `VERSION` file (e.g. https://github.com/gardener/gardener/commit/e75d09648f3a1d95da4aa5aca121e08c726a95e4).
In your case, the changes in the `Chart.yaml` should become part of such a "release commit".

Comment (morremeyer) [reply docs/development/process.md:49]: I tried to find out what triggers the „release commit“ in g/g, but I can’t find it. As it’s the bot, I suspected it was an action, but there’s no matching one.

Did I miss something? Can you point me to where I need to incorporate that change?

Comment (rfranzke) [reply docs/development/process.md:49]: > I tried to find out what triggers the „release commit“ in g/g,

This is implemented deep down in the https://github.com/gardener/cc-utils repository, @ccwienk or @AndreasBurger can explain better.

> Can you point me to where I need to incorporate that change?

The only thing you need to provide is such a `prepare_release` script which performs your changes. Once the PR is merged, a Gardener maintainer has to enhance the pipeline definitions to make the CI/CD system executing your `prepare_release` script when a new Gardener release is triggered.

Comment (morremeyer) [reply docs/development/process.md:49]: Do I understand it correctly that those scripts are run in an alpine docker container? If not, which environment do I have to assume?

Comment (rfranzke) [reply docs/development/process.md:49]: Yes, I think it's `alpine`. Not sure if this is configurable meanwhile.

File: docs/development/process.md
Comment (schrodit) [inline docs/development/process.md:48]: Not sure if I understand it correctly.
In the first statement you say that the versions should be bumped during the release and set to the released version.

But then the upload/release of the helm chart is done on every head update? So also for every dev version? What about hotfixes?

> Set the version to the same as the gardener version that is to be released, but remove the `v` in front of the version number for a valid semver version.

Can you specify what version is meant. Should it be the chart version or the version of the oci images inside the helm chart.

**Comment by timebertt** [comment]
Please wait with this PR, I have some feedback on the approach.
Will try to comment somewhen today.

**Review by timebertt — CHANGES_REQUESTED** [review]
See https://github.com/gardener/gardener/issues/4123#issuecomment-868476298.
Let's please not add GH pages helm repos in every of our repos that publishes helm charts and rather sync/push them to a dedicated repo.

**Comment by morremeyer** [comment]
I will change the approach here according to [@timebertt’s comment](https://github.com/gardener/gardener/issues/4123#issuecomment-868476298) in #4123.

**Pull request closed without merging**
